csrc/device_lower/utils.cpp

```diff
@@ -1013,6 +1016,552 @@ bool predicateAtEnd(ForLoop* loop) {
   return true;
 }
 
+// Implementation of:
+//   Val* proveLinearAndGetStride(
+//       const ValGraph& id_graph,
+//       const ValGroup& linear_g,
+//       const ValGroups& domain);
+//
+// The idea is to propagate from linear_g to domain, keep track of how linear_g
+// lives in domain. For example, let's consider the following two schedules
+// (same example as the NVFuserTest.ProveLinearAndGetStride test):
+//
+// v1:
+//        I0         I1
+//       /  \       /  \.
+//          128        128
+//          / \        / \.
+//         /   \      /   \.
+//        /     \    /     \.
+//       /       \  /       \.
+//      /         \/        64.
+//     /          /\       /  \.
+//    /          /  \     /    \.
+//   16        [2]   8   8      8
+//                    \ /
+//                    xor
+//                    / \.
+//                   8   8
+//
```

Comment:
Correct me if this understanding is wrong:

```
Inner split of I0 by 128 -> I0o, 128_1
Inner split of I1 by 128 -> I1o, 128_2
Inner split of 128_1 by 8 -> 16, 8_1
Inner split of 128_2 by 64 -> [2], 64
Inner split of 64 by 8 -> 8_2, 8_3
Xor swizzle of 8_1 and 8_2 -> 8_4, 8_5
```

(Used `_` to denote different IDs with the same extent)

Reply:
Yes, exactly